rebuild/src/addon.h

```diff
@@ -0,0 +1,177 @@
+#ifndef BLST_TS_ADDON_H__
+#define BLST_TS_ADDON_H__
+
+#include <openssl/rand.h>
+
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <string_view>
+
+#include "blst.hpp"
+#include "napi.h"
+
+// TODO: these should come out post PR review
+using std::cout;
+using std::endl;
+
+#define BLST_TS_RANDOM_BYTES_LENGTH 8U
+
+#define BLST_TS_FUNCTION_PREAMBLE(info, env, module)                           \
```

Comment:
Can those macros be functions (potentially templated)? Macros tend to be quite difficult to debug, and may increase the binary size (all code is inlined). It also makes static checks more difficult.

Reply:
I actually wanted that code inlined.  Its boilerplate but needs to happen at the top of each function.  Same with the unwrapping of args.  I had it as a class but it increased runtime size so I changed them to macros.  Its how most of the libs do these types of things so would call it "native node industry practice" but to be fair most devs that write this type of code do not come from a c++ background ssooooooo......